### PR TITLE
Api 1030 medication request controller from medication order

### PR DESCRIPTION
**Pull request by taylorrassmannlibertyits**
https://vajira.max.gov/browse/API-1030

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/MedicationRequestFromMedicationOrderRepository.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package gov.va.api.health.dataquery.service.controller.medicationrequest;
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/MedicationRequestFromMedicationOrderRepository.java:1]: I think this should be deleted and just use the existing `MedicationOrderRepository`.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestController.java [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+              .add("page", page)
+              .add("_count", count)
+              .build(),
+          count,
+          repository.findByIcn(
+              icn,
+              PageRequest.of(
+                  page - 1, count == 0 ? 1 : count, MedicationOrderEntity.naturalOrder())));
+    } else {
+      return MedicationRequest.Bundle.builder().build();
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestController.java:209]: too empty, it should have the paging links etc. just have empty `entry[]` field.

DR isn't exactly the same as the other resources, but the basic concept is here: https://github.com/department-of-veterans-affairs/health-apis-data-query/blob/master/data-query/src/main/java/gov/va/api/health/dataquery/service/controller/diagnosticreport/Dstu2DiagnosticReportController.java#L249

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@DataJpaTest
+@RunWith(SpringRunner.class)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+public class R4MedicationRequestControllerTest {
```
Comment (bschofield-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestControllerTest.java:38]: Need a test for `intent=<something not order>` to assert that a valid, but empty bundle is returned.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/MedicationRequestFromMedicationOrderRepository.java (hunk 1 above)
Comment (taylorrassmannlibertyits) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/MedicationRequestFromMedicationOrderRepository.java:1]: Updated in next push.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestController.java (hunk 2 above)
Comment (taylorrassmannlibertyits) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestController.java:209]: It makes so much more sense now. Thanks for the example. Updated in next push.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationOrderTransformer.java [hunk 4]
```diff
@@ -183,15 +185,24 @@ static Timing timing(
         .build();
   }
 
+  Extension requesterExtension(DatamartReference requester) {
```
Comment (ilaflamme-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationOrderTransformer.java:188]: Can this guy get a unit test?
